File: GymFlow-Web/src/api/settings/systemConfig.ts
import request from '@/utils/request'
import type { ApiResponse } from '@/types/common'

// 基本设置DTO
export interface BasicConfigDTO {
  systemName: string
  systemLogo?: string
}

// 业务设置DTO
export interface BusinessConfigDTO {
  businessStartTime: string
  businessEndTime: string
  courseRenewalDays: number
  courseCancelHours: number
  minClassSize: number
  maxClassCapacity: number
}

// 系统配置响应DTO
export interface SystemConfigResponseDTO {
  basic: BasicConfigDTO
  business: BusinessConfigDTO
  updateTime: string
}

// 系统配置更新请求
export interface SystemConfigUpdateRequest {
  basic: BasicConfigDTO
  business: BusinessConfigDTO
}

// 系统配置API
export const systemConfigApi = {
  /**
   * 获取系统配置
   */
  getConfig(): Promise<ApiResponse<SystemConfigResponseDTO>> {
    return request({
      url: '/settings/systemConfig',
      method: 'GET'
    })
  },

  /**
   * 更新系统配置
   */
  updateConfig(data: SystemConfigUpdateRequest): Promise<ApiResponse> {
    return request({
      url: '/settings/systemConfig',
      method: 'PUT',
      data
    })
  },

  /**
   * 重置系统配置
   */
  resetConfig(): Promise<ApiResponse> {
    return request({
      url: '/settings/systemConfig/reset',
      method: 'POST'
    })
  },

  /**
   * 上传Logo
   */
  uploadLogo(file: File): Promise<ApiResponse<{ url: string }>> {
    const formData = new FormData()
    formData.append('file', file)
    return request({
      url: '/upload/logo',
      method: 'POST',
      data: formData,
      headers: {
        'Content-Type': 'multipart/form-data'
      }
    })
  }
}

File: GymFlow-Backend/src/main/java/com/gymflow/entity/settings/SystemConfig.java
package com.gymflow.entity.settings;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Data
@TableName("system_config")
public class SystemConfig {

    @TableId(type = IdType.AUTO)
    private Long id;

    // ========== 基本设置 ==========

    /**
     * 系统名称
     */
    @TableField("system_name")
    private String systemName;

    /**
     * 系统logo URL
     */
    @TableField("system_logo")
    private String systemLogo;

    // ========== 业务设置 ==========

    /**
     * 营业开始时间
     */
    @TableField("business_start_time")
    private LocalTime businessStartTime;

    /**
     * 营业结束时间
     */
    @TableField("business_end_time")
    private LocalTime businessEndTime;

    /**
     * 课程提前续约时间（天）
     */
    @TableField("course_renewal_days")
    private Integer courseRenewalDays;

    /**
     * 课程取消时间限制（小时）
     * 课程开始前几小时内可以取消
     */
    @TableField("course_cancel_hours")
    private Integer courseCancelHours;

    /**
     * 最低开课人数（团课）
     */
    @TableField("min_class_size")
    private Integer minClassSize;

    /**
     * 最大课程容量（团课最多参与人数）
     */
    @TableField("max_class_capacity")
    private Integer maxClassCapacity;

    /**
     * 最后更新时间
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}